src/pretix/api/serializers/event.py

```diff
@@ -989,6 +989,40 @@ def prefetch_by_id(items, qs, id_attr, target_attr):
             setattr(item, target_attr, result.get(getattr(item, id_attr)))
 
 
+class SeatBulkBlockInputSerializer(serializers.Serializer):
+    ids = serializers.ListField(child=serializers.IntegerField(), required=False, allow_empty=True)
+    seat_guids = serializers.ListField(child=serializers.CharField(), required=False, allow_empty=True)
+
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+
+        if data.get("seat_guids") and data.get("ids"):
+            raise ValidationError("Please pass either seat_guids or ids.")
+
+        if data.get("seat_guids"):
+            seat_ids = data["seat_guids"]
+            if len(seat_ids) > 10000:
+                raise ValidationError({"seat_guids": ["Please do not pass over 10000 seats."]})
+
+            seats = {s.seat_guid: s for s in self.context["queryset"].filter(seat_guid__in=seat_ids)}
+            for s in seat_ids:
+                if s not in seats:
+                    raise ValidationError({"seat_guids": [f"The seat '{s}' does not exist."]})
+        elif data.get("ids"):
+            seat_ids = data["ids"]
+            if len(seat_ids) > 10000:
+                raise ValidationError({"ids": ["Please do not pass over 10000 seats."]})
+
+            seats = self.context["queryset"].in_bulk(seat_ids)
+            for s in seat_ids:
+                if s not in seats:
+                    raise ValidationError({"ids": [f"The seat '{s}' does not exist."]})
+        else:
+            raise ValidationError("Please pass either seat_guids or ids.")
```

Comment:
Would it make sense to deduplicate code here?

```suggestion
        if data.get("seat_guids"):
            ids_key = "seat_guids"
        elif data.get("ids"):
            ids_key = "ids"
        else:
            raise ValidationError("Please pass either seat_guids or ids.")

        seat_ids = data[ids_key]
        if len(seat_ids) > 10000:
            raise ValidationError({ids_key: ["Please do not pass over 10000 seats."]})

        if ids_key == "seat_guids":
            seats = {s.seat_guid: s for s in self.context["queryset"].filter(seat_guid__in=seat_ids)}
        else:
            seats = self.context["queryset"].in_bulk(seat_ids)

        for s in seat_ids:
            if s not in seats:
                raise ValidationError({ids_key: [f"The seat '{s}' does not exist."]})
```

Reply:
It's less duplicate, true, but exactly the same length and I don't really find it easier to read